alf/examples/metadrive/ppg_metadrive_transformer_conf.py

```diff
@@ -61,57 +80,85 @@ class ObservationCombiner(torch.nn.Module):
 
     """
 
-    def __init__(self, d_map_feature: int, d_ego_feature: int,
-                 d_agent_feature: int, d_model: int):
+    def __init__(self, map_feature: EmbeddingConfig,
+                 ego_feature: EmbeddingConfig, agent_feature: EmbeddingConfig,
+                 d_model: int):
         super().__init__()
 
-        self._map_fc = alf.layers.FC(
-            d_map_feature, d_model, activation=torch.relu_)
-
-        self._agent_fc = alf.layers.FC(
-            d_agent_feature, d_model, activation=torch.relu_)
-
-        self._ego_fc = alf.layers.FC(
-            d_ego_feature, d_model, activation=torch.relu_)
+        self._map_eb = map_feature.make_embedding_net(d_model)
+        self._agent_eb = agent_feature.make_embedding_net(d_model)
+        self._ego_eb = ego_feature.make_embedding_net(d_model)
 
     def forward(self, inputs):
         map_sequence, ego, agents = inputs
 
-        x0 = self._ego_fc(ego).unsqueeze(1)  # [B, 1, d_model]
+        x0 = self._ego_eb(ego).unsqueeze(1)  # [B, 1, d_model]
 
         # The input ``sequence`` is [B, L, d_map_feature]
-        x1 = self._map_fc(map_sequence)  # [B, L, d_model]
+        x1 = self._map_eb(map_sequence)  # [B, L, d_model]
 
         x2 = agents.view(*agents.shape[:2], -1)
-        x2 = self._agent_fc(x2)  # [B, A, d_model]
+        x2 = self._agent_eb(x2)  # [B, A, d_model]
 
         return torch.cat([x0, x1, x2], dim=1)
 
 
+class MaskedTransformer(torch.nn.Module):
+    def __init__(self, d_model, num_heads, num_layers, map_limit, agent_limit):
+        super().__init__()
+        self._memory_size = 1 + map_limit + agent_limit
+
+        tf_layers = []
+        for i in range(num_layers):
+            tf_layers.append(
+                alf.layers.TransformerBlock(
+                    d_model=d_model,
+                    num_heads=num_heads,
+                    memory_size=self._memory_size,
+                    positional_encoding='none'))
+        self._tf_layers = torch.nn.ModuleList(tf_layers)
+
+    def forward(self, inputs):
+        x, map_mask, agent_mask = inputs
+        B = x.shape[0]
+        mask = torch.hstack((torch.ones(B, 1, dtype=bool), map_mask,
```

Comment:
torch.ones => torch.zeros

Reply:
Thanks for pointing out this bad bug. Fixed.
 I'll run the training again to verify before submitting the PR.